src/web/gwc/src/main/java/org/geoserver/gwc/web/layer/UnconfiguredCachedLayersProvider.java

```diff
@@ -74,7 +75,8 @@
      * Simple cache for the last computed size based on the keywords, which in turn are the sole
      * input for the filter (at the moment, at least)
      */
-    private class CachedSize {
+    private class CachedSize implements Serializable {
```

Comment:
Yep, I see how this is happening. At the same time, I haven't noticed that trace in the logs, guess my normal usage does not force Wicket to serialize pages (grepped through the logs of the data dirs I use for testing, it's not there...).

Can you think of a way to make these easier to catch when developing? Some settings in the WicketTestSupport that would make these appear right away?

Reply:
that's an interesting thought, I'm not sure what's triggering it in my case but probably the use of `WebUIMode.DO_NOT_REDIRECT`. Won't make a difference in the simple "check page loads" test though.

The following test can be added, but not sure how/if to enforce it for all wicket tests:
```
    @Test
    public void testSerializable() throws IOException {
        login();
        NewCachedLayerPage page = new NewCachedLayerPage();
        ISerializer serializer = getGeoServerApplication().getFrameworkSettings().getSerializer();
        //this would only result in an ERROR level log
        serializer.serialize(page);
        
        try {
            new ObjectOutputStream(new ByteArrayOutputStream()).writeObject(page);
        }catch(NotSerializableException e) {
            fail(e.getMessage());
        }
    }
```
```
java.lang.AssertionError: org.geoserver.gwc.web.layer.UnconfiguredCachedLayersProvider$CachedSize
```

Reply:
Actually I just did add the test for this specific case and force-pushed.

Reply:
Hum... looking at the test you setup there gave me an idea... the simple "check page loads" code goes though ``WicketTester.startPage``.... so maybe we could roll a subclass of it, that checks the page can be serialized, whenever startPage is used... yes, it would do that multiple times for nothing, but on the other end, it's too easy to forget doing so manually for each and every page.

Maybe the subclass could be smart enough to keep a cache of which classes it has already verified.

It still would not cover all cases, maybe the page could get something non serializable in its model after some user interaction... oh well, just  thinking out loud...